pkg/volume/azure_dd/mounter.go

```
@@ -0,0 +1,181 @@
+
+func (m *azureDiskMounter) GetAttributes() volume.Attributes {
+	volumeSource, _ := getVolumeSource(m.spec)
+	return volume.Attributes{
+		ReadOnly:        *volumeSource.ReadOnly,
+		SupportsSELinux: true,
+	}
+}
+
+func (m *azureDiskMounter) CanMount() error {
+	return nil
+}
+
+func (m *azureDiskMounter) SetUp(fsGroup *int64) error {
+	return m.SetUpAt(m.GetPath(), fsGroup)
+}
+
+func (m *azureDiskMounter) GetPath() string {
+	return getPath(m.dataDisk.podUID, m.dataDisk.volumeName, m.plugin.host)
+}
+
+func (m *azureDiskMounter) SetUpAt(dir string, fsGroup *int64) error {
+	mounter := m.plugin.host.GetMounter()
+	volumeSource, err := getVolumeSource(m.spec)
+
+	if err != nil {
+		glog.Infof("azureDisk - mounter failed to get volume source for spec %s", m.spec.Name())
+		return err
+	}
+
+	diskName := volumeSource.DiskName
+	mountPoint, err := mounter.IsLikelyNotMountPoint(dir)
+
+	if err != nil && !os.IsNotExist(err) {
+		glog.Infof("azureDisk - cannot validate mount point for disk %s on  %s %v", diskName, dir, err)
+		return err
+	}
+	if !mountPoint {
+		glog.Infof("azureDisk - Not a mounting point for disk %s on %s %v", diskName, dir, err)
+		return nil
+	}
+
+	if err := os.MkdirAll(dir, 0750); err != nil {
+		glog.Infof("azureDisk - mkdir failed on disk %s on dir: %s (%v)", diskName, dir, err)
+		return err
+	}
+
+	options := []string{"bind"}
+
+	if *volumeSource.ReadOnly {
+		options = append(options, "ro")
+	}
+
+	glog.V(4).Infof("azureDisk - Attempting to mount %s on %s", diskName, dir)
+	isManagedDisk := (*volumeSource.Kind == v1.AzureManagedDisk)
+	globalPDPath, err := makeGlobalPDPath(m.plugin.host, volumeSource.DataDiskURI, isManagedDisk)
+
+	if err != nil {
+		return err
+	}
+
+	err = mounter.Mount(globalPDPath, dir, *volumeSource.FSType, options)
+	if err != nil {
+		mountPoint, err := mounter.IsLikelyNotMountPoint(dir)
+		if err != nil {
+			glog.Infof("azureDisk - IsLikelyNotMountPoint check failed: %v", err)
+			return err
```

nit: please return an error list here so the user sees all errors.